Autonomous trajectory code for Galactic Search challenge (only drive, no intake)

mukundkk opened the pull request.

VaibhavKaranam committed:
Update Constants.java

VaibhavKaranam committed:
Update and rename ExampleCommand.java to ServoControlCommand.java

VaibhavKaranam committed:
Create RunServo

VaibhavKaranam committed:
Rename RunServo to RunServo.java

VaibhavKaranam committed:
Create TeleopArcadeDriveCommand.java

VaibhavKaranam committed:
Create TeleopTankDriveCommand.java

VaibhavKaranam committed:
Update and rename ExampleSubsystem.java to DriveSubsystem.java

VaibhavKaranam committed:
Create ServoSubsystem.java

VaibhavKaranam committed:
Create TestMotorSubsystem.java

VaibhavKaranam committed:
Update Robot.java

VaibhavKaranam committed:
Update RobotContainer.java

VaibhavKaranam committed:
Create TestMotorCommand

VaibhavKaranam committed:
Create TestMotorByTimeCommand.java

VaibhavKaranam committed:
Rename TestMotorCommand to TestMotorCommand.java

VaibhavKaranam committed:
Create TestMotorEncoderCommand.java

Euna4Ibmcloud committed:
Updated Constants.java

Euna4Ibmcloud committed:
Updated Constants.java

Euna4Ibmcloud committed:
Updated RobotContainer.java

Euna4Ibmcloud committed:
Updated DriveSubsystem.java

Euna4Ibmcloud committed:
Updated Constants.java

Euna4Ibmcloud committed:
Updated RobotContainer.java

Euna4Ibmcloud committed:
Updated ServoControlCommand.java

Euna4Ibmcloud committed:
Updated RobotContainer.java

Euna4Ibmcloud committed:
Updated Constants.java and Robot.java

Euna4Ibmcloud committed:
Updated RobotContainer.java

Euna4Ibmcloud committed:
Updated Robot.java

Euna4Ibmcloud committed:
Deleted unnecessary files

Euna4Ibmcloud committed:
Stop motors in init

Euna4Ibmcloud committed:
Added previous copyright statement

Euna4Ibmcloud committed:
New RobotContainer methods, objects aren't exposed

Euna4Ibmcloud committed:
Created method to stop motors in init

Euna4Ibmcloud committed:
Removed unused imports and commented out auto code

RahulBansal1112 committed:
added gyro, odometry, and methods that will help in creating trajectories

mukundkk committed:
Write autonomous trajectory code for Galactic Search

Use WPILib trajectory classes to write autonomous trajectory code for
the Galactic Search challenge, Path B (red markers).

Note: the feedforward, feedback, and trackwidth constants should
be verified and/or revised after running the drivetrain
characterization routine another time.

AG6GR commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -2,52 +2,75 @@
+public static final int SPARK_BACK_RIGHT_ID = 2;
+
+// units: meters 
+public static final double kSDrive = 0.183;
```
Standard practice is to name constants all caps and underscore separated.

AG6GR commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -79,9 +92,65 @@ private void configureButtonBindings() {
-		return null; // no auto command yet
+		// create a voltage constraint to ensure we don't accelerate too fast
+		// set maximum voltage to 10V rather than 12V to allow for voltage sag during operation
+		var autoVoltageConstraint = 
```
Please define the type of all variables.

AG6GR commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -2,52 +2,75 @@
+public static final double DRIVE_I = 0;
+public static final double DRIVE_D = 0;
+
+public static final double DRIVE_TICKS_PER_INCH = 1;
```
This doesn't feel right. You should derive from DRIVE_METERS_PER_ROTATION and getEncoder().getPositionConversionFactor()

AG6GR commented on src/main/java/frc/robot/subsystems/DriveSubsystem.java:
```diff
@@ -1,106 +1,176 @@
+        return frontRightMotor.getEncoder().getPosition();
+    }
+
+    public void setMaxOutput(double maxOutput) {
```
Do we ever call setMaxOutput? We should on init.

AG6GR reviewed (changes requested):
It looks like there are some merge conflicts, do see if you can resolve them.

mukundkk committed:
Address PR comments

Change naming of all new drive constants to caps with underscore
separation. Call setMaxOutput() in constructor of DriveSubsystem.
Define type of voltage constraint variable. Correct derivation
of drive ticks per inch.

mukundkk committed:
Update drive constants after recharacterization

mukundkk committed:
Add test trajectory and comment out main one

Create and use basic trajectory for testing trajectory following.
Comment out full path and add Pose2d version of point D5 for
testing.
